pkg/kubelet/util/util_unix_test.go

```diff
@@ -133,3 +133,41 @@ func TestIsUnixDomainSocket(t *testing.T) {
 		assert.Equal(t, result, test.expectSocket, "Unexpected result from IsUnixDomainSocket: %v for %s", result, test.label)
 	}
 }
+
+func TestGetAddressAndDialer(t *testing.T) {
+	tests := []struct {
+		endpoint     string
+		expectError  bool
+		expectedAddr string
+	}{
+		{
+			endpoint:     "unix:///tmp/s1.sock",
+			expectError:  false,
+			expectedAddr: "/tmp/s1.sock",
+		},
+		{
+			endpoint:     "unix:///tmp/f6.sock",
+			expectError:  false,
+			expectedAddr: "/tmp/f6.sock",
+		},
+		{
+			endpoint:    "tco:/localhost:9090",
+			expectError: true,
+		},
+		{
+			// The misspelling is intentional to make it error
+			endpoint:    "htta://free-test.com",
+			expectError: true,
+		},
```

Comment:
could you add a test case for a valid but not unix domain address so we exercise that error case?

Reply:
> could you add a test case for a valid but non-unix domain address so we exercise that error case?
> https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/util/util_unix.go#L87-L89

Of course, I will add it immediately😊